### PR TITLE
feat(web-app): add divider component

### LINKED ISSUE #777: Create list / section / divider component
Turn these custom styles into a component we can reuse across pages.

Look at bringing over [ExpressiveList](https://gatsby-theme-carbon.vercel.app/components/ExpressiveList) or [ExpressiveListContainer](https://gatsby-theme-carbon.vercel.app/components/ExpressiveListContainer) from Gatsby theme to use here possibly?

This is needed on How Carbon works, Meetups, Asset landing, roadmap

- [ ] Document component in Storybook

Screen Shot 2022-05-24 at 1 34 19 PM
Screen Shot 2022-05-24 at 1 57 41 PM
Screen Shot 2022-05-24 at 1 57 58 PM

Add new component to

- [ ] storybook
- [ ] http://localhost:3000/about-carbon/platform-roadmap
- [ ] http://localhost:3000/about-carbon/meetups

### PR DESCRIPTION
**Pull request by andreancardona**
Closes #777 

### TEST:

**Divider component in action:** 

1.
- http://localhost:3000/about-carbon/meetups
- Make sure it matches the margin / padding: https://next.carbondesignsystem.com/about-carbon/platform-roadmap
2.
- http://localhost:3000/about-carbon/platform-roadmap
- Make sure it matches the margin / padding:  https://next.carbondesignsystem.com/about-carbon/platform-roadmap

**Divider component:** 
- http://localhost:3000/mdx/divider
- https://gatsby-theme-carbon.vercel.app/components/ExpressiveList

**Storybook:**
- http://localhost:6006/?path=/docs/mdx-divider--default

### CODE REVIEW

File: services/web-app/components/expressive-list/expressive-list.module.scss [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  padding-bottom: spacing.$spacing-10;
+  border-bottom: 1px solid theme.$layer-accent;
+  background: theme.$layer-01;
+
+  @include breakpoint.breakpoint-down(lg) {
+    padding-left: spacing.$spacing-05;
+  }
+}
+
+:global(.cds--subgrid .cds--subgrid--wide) {
```
Comment (francinelucca) [inline services/web-app/components/expressive-list/expressive-list.module.scss:23]:
```suggestion
// carbon override subgrid
:global(.cds--subgrid .cds--subgrid--wide) {
```

**Review by francinelucca — COMMENTED** [review]
minor comment, also, should we add this to `index.mdx` and add a storybook as well?

File: services/web-app/components/divider/divider.module.scss [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+@use '@carbon/react/scss/spacing' as spacing;
+@use '@carbon/react/scss/theme' as theme;
+@use '@carbon/react/scss/type' as type;
+
+.list-row {
+  padding: spacing.$spacing-07 0 calc(spacing.$spacing-09 + spacing.$spacing-03);
+  border-bottom: 1px solid theme.$border-strong-01;
+}
+
+:global(.cds--subgrid--wide) {
```
Comment (francinelucca) [inline services/web-app/components/divider/divider.module.scss:18]:
```suggestion
// override carbon grid
:global(.cds--subgrid--wide) {
```

File: services/web-app/pages/mdx/index.mdx [hunk 3]
```diff
@@ -141,7 +141,7 @@ platea dictumst.
   </ArticleCard>
 </Column>
 <Column colMd={4} colLg={4} noGutterMdLeft>
-    <ArticleCard
+    <
```
Comment (francinelucca) [inline services/web-app/pages/mdx/index.mdx:144]: what's happening here? 😅  this is breaking the mdx page for me 
image

File: services/web-app/stories/mdx/divider.stories.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+ */
+
+import { Column, Grid } from '@carbon/react'
+
+import Divider from '@/components/divider'
+
+const stories = {
+  title: 'MDX/Divider',
+  component: Divider
+}
```
Comment (francinelucca) [inline services/web-app/stories/mdx/divider.stories.js:15]: Let's hide these controls, they don't make much sense? 
image

```suggestion
  title: 'MDX/Divider',
  component: Divider,,
  argTypes: {
    children: {
      control: false
    },
    className: {
      control: false
    }
  }
}
```

**Review by francinelucca — CHANGES_REQUESTED** [review]
looks good visually.  some comments + need to fix CI checks

**Review by alisonjoseph — CHANGES_REQUESTED** [review]
I think you need to run `npm run lint:fix` for the linting error

File: services/web-app/components/divider/divider.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import clsx from 'clsx'
+import PropTypes from 'prop-types'
+
+import * as styles from './divider.module.scss'
+
```
Comment (alisonjoseph) [inline services/web-app/components/divider/divider.js:12]: You'll want to add a description here that will show up in storybook

```
/**
 * The `<Divider>` component is a wrapper component that adds a top border
 * and spacing to divide sections of content.
 */
```

Or something like that

File: services/web-app/components/divider/divider.module.scss [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+ */
+
+@use '@carbon/react/scss/breakpoint' as breakpoint;
+@use '@carbon/react/scss/spacing' as spacing;
+@use '@carbon/react/scss/theme' as theme;
+@use '@carbon/react/scss/type' as type;
+
+.divider {
+  padding: spacing.$spacing-07 0 calc(spacing.$spacing-09 + spacing.$spacing-03);
+  border-top: 1px solid theme.$border-strong;
```
Comment (alisonjoseph) [inline services/web-app/components/divider/divider.module.scss:15]:
```suggestion
  padding-top: spacing.$spacing-07;
  border-top: 1px solid theme.$border-strong;
  margin-top: calc(spacing.$spacing-09 + spacing.$spacing-03);
```

suggestion to get correct spacing above the first element

**Comment by alisonjoseph** [comment]
@andreancardona I added this component to the meetups page to test and pushed to this branch http://localhost:3000/about-carbon/meetups

File: services/web-app/pages/mdx/index.mdx (hunk 3 above)
Comment (alisonjoseph) [inline services/web-app/pages/mdx/index.mdx:144]:
```suggestion
    <ArticleCard
```

**Review by francinelucca — COMMENTED** [review]
1 - Border top seems longer in width in next.carbondesignsystem.com , are we ok with that? 
2 - Do we want to add this to http://localhost:3000/about-carbon/how-carbon-works, http://localhost:3000/assets as well?

other than that lgtm

File: services/web-app/components/divider/divider.module.scss [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+ * LICENSE file in the root directory of this source tree.
+ */
+
+@use '@carbon/react/scss/spacing' as spacing;
+@use '@carbon/react/scss/theme' as theme;
+
+.divider {
+  padding-top: spacing.$spacing-04;
+  border-top: 1px solid theme.$border-strong;
+  margin-bottom: calc(spacing.$spacing-09 + spacing.$spacing-03);
```
Comment (alisonjoseph) [inline services/web-app/components/divider/divider.module.scss:14]:
```suggestion
  padding: spacing.$spacing-04 spacing.$spacing-05 0 0;
  border-top: 1px solid theme.$border-strong;
  margin-right: -(spacing.$spacing-05);
  margin-bottom: calc(spacing.$spacing-09 + spacing.$spacing-03);
```

@francinelucca with the eagle eyes! you're correct its 1rem off on the right, this should fix. 

It wouldn't let me do shorthand for margin with the negative value in the middle 🤷‍♀️

**Review by alisonjoseph — CHANGES_REQUESTED** [review]
LGTM other than the right margin/padding Francine pointed out

I added and pushed to http://localhost:3000/about-carbon/how-carbon-works

**Review by alisonjoseph — APPROVED** [review]

File: services/web-app/components/divider/divider.module.scss [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+@use '@carbon/react/scss/spacing' as spacing;
+@use '@carbon/react/scss/theme' as theme;
+@use '@carbon/react/scss/breakpoint' as breakpoint;
+
+.divider {
+  padding-top: spacing.$spacing-03;
+  border-top: 1px solid theme.$border-strong;
+  margin-top: calc(spacing.$spacing-09 + spacing.$spacing-03);
+
+  @include breakpoint.breakpoint(md) {
```
Comment (alisonjoseph) [inline services/web-app/components/divider/divider.module.scss:17]:
```suggestion
  @include breakpoint.breakpoint(lg) {
```

**Review by francinelucca — APPROVED** [review]
💃

**Pull request merged**